framework/tlvf/src/include/tlvf/CmduMessageTx.h

```diff
@@ -25,7 +25,7 @@ class CmduMessageTx : public CmduMessage {
     std::shared_ptr<cCmduHeader> load();
     std::shared_ptr<tlvVendorSpecific> add_vs_tlv(tlvVendorSpecific::eVendorOUI voui);
     template <class T> std::shared_ptr<T> addClass() { return msg.addClass<T>(); }
-
+    void reset() { msg.reset(false); }
```

Comment:
What you wrote in the commit message is not true: 
We do not send the cmdu twice. If the we are on certification, we hold the cmdu without sending it and waiting for the ucc_listener to send it, otherwise, we send it.
At this point in the review I can't see how you use it yet, but the assumption on the commit message that we send the cmdu twice, makes me think that this commit should be dropped.

Reply:
This commit is necessary even if we don't send it twice - since it is used by later commit to reset the certification cmdu which is like "freeing" it so it won't be used endlessly.

Regarding the sending of the same cmdu twice - I disagree - @arnout and I discussed this and came to the conclusion that we should not behave differently in certification mode.
This means that on every flow we should proceed as usual, and in the flows where we need to send a CMDU at a later time for the UCC sniffer to catch it, we will just send it a second time.

@arnout Please correct me if I'm wrong.

Reply:
OK so apparently I *was* wrong. Fixed this as @morantr and @arnout suggested.